bridge/gitlab/import.go

```diff
@@ -0,0 +1,351 @@
+				keyGitlabUrl: issue.WebURL,
+			},
+		)
+
+		if err != nil {
+			return nil, err
+		}
+
+		// importing a new bug
+		gi.importedIssues++
+
+		return b, nil
+	}
+
+	return b, nil
+}
+
+func (gi *gitlabImporter) ensureNote(repo *cache.RepoCache, b *cache.BugCache, note *gitlab.Note) error {
+	id := parseID(note.ID)
+
+	// ensure issue author
+	author, err := gi.ensurePerson(repo, note.Author.ID)
+	if err != nil {
+		return err
+	}
+
+	hash, errResolve := b.ResolveOperationWithMetadata(keyGitlabId, id)
+	if errResolve != cache.ErrNoMatchingOp {
+		return err
+	}
+
+	noteType, body := GetNoteType(note)
+	switch noteType {
+	case NOTE_CLOSED:
+		_, err = b.CloseRaw(
+			author,
+			note.CreatedAt.Unix(),
+			map[string]string{
+				keyGitlabId:  id,
+				keyGitlabUrl: "",
+			},
+		)
+		return err
+
+	case NOTE_REOPENED:
+		_, err = b.OpenRaw(
+			author,
+			note.CreatedAt.Unix(),
+			map[string]string{
+				keyGitlabId:  id,
+				keyGitlabUrl: "",
+			},
+		)
+		return err
+
+	case NOTE_DESCRIPTION_CHANGED:
+		issue := gi.iterator.IssueValue()
+
+		// since gitlab doesn't provide the issue history
+		// we should check for "changed the description" notes and compare issue texts
+		// TODO: Check only one time and ignore next 'description change' within one issue
+		if issue.Description != b.Snapshot().Comments[0].Message {
+
+			// comment edition
+			_, err = b.EditCommentRaw(
+				author,
+				note.UpdatedAt.Unix(),
+				target,
+				issue.Description,
+				map[string]string{
+					keyGitlabId:  id,
+					keyGitlabUrl: "",
+				},
+			)
+
+			return err
+
+		}
+
+	case NOTE_COMMENT:
+
+		cleanText, err := text.Cleanup(body)
+		if err != nil {
+			return err
+		}
+
+		// if we didn't import the comment
+		if errResolve == cache.ErrNoMatchingOp {
+
+			// add comment operation
+			_, err = b.AddCommentRaw(
+				author,
+				note.CreatedAt.Unix(),
+				cleanText,
+				nil,
+				map[string]string{
+					keyGitlabId:  id,
+					keyGitlabUrl: "",
+				},
+			)
+
+			return err
+		}
+
+		// if comment was already exported
+
+		// if note wasn't updated
+		if note.UpdatedAt.Equal(*note.CreatedAt) {
+			return nil
+		}
+
+		// search for last comment update
+		comment, err := b.Snapshot().SearchComment(hash)
+		if err != nil {
+			return err
+		}
+
+		// compare local bug comment with the new note body
+		if comment.Message != cleanText {
+			// comment edition
+			_, err = b.EditCommentRaw(
+				author,
+				note.UpdatedAt.Unix(),
+				target,
+				cleanText,
+				map[string]string{
+					// no metadata unique metadata to store
+					keyGitlabId:  "",
+					keyGitlabUrl: "",
+				},
+			)
+
+			return err
+		}
+
+		return nil
+
+	case NOTE_TITLE_CHANGED:
+		// title change events are given new notes
+		_, err = b.SetTitleRaw(
+			author,
+			note.CreatedAt.Unix(),
+			body,
+			map[string]string{
+				keyGitlabId:  id,
+				keyGitlabUrl: "",
+			},
+		)
+
+		return err
+
+	case NOTE_UNKNOWN:
+		//TODO: send warning via channel
```

Comment:
no need for a message, it just means gitlab have a new kind of not. It's not the user's problem.